nix/forVscodeVersionRaw.nix

```diff
@@ -0,0 +1,207 @@
+{
+  pkgs,
+  dataPath ? if isOpenVsx then ../data/extensions_openvsx.json else ../data/extensions.json,
+  version ? pkgs.vscode.version,
+  extensions,
+  pickPreRelease ? false,
+  decorators ? null,
+  isOpenVsx ? false,
+}:
+let
+  inherit (pkgs) lib system;
+  allExtensions = lib.importJSON dataPath;
+  platformMap = {
+    x86_64-linux = "linux-x64";
+    i686-linux = "linux-x64";
+    aarch64-linux = "linux-arm64";
+    armv7l-linux = "linux-armhf";
+    x86_64-darwin = "darwin-x64";
+    aarch64-darwin = "darwin-arm64";
+  };
+  platform = platformMap.${system} or null;
+
+  # Get an attrset of extensions and their wanted versions
+  wantedExts = builtins.listToAttrs (lib.map splitExt extensions);
+
+  /**
+    Split an extension name into its parts and return a name-value pair.
+    The name is the first two parts of the extension name, and the value is the version.
+
+    # Type
+
+    ```
+    splitExt :: string -> { name :: string; value :: string; }
+    ```
+
+    # Example
+
+    ```nix
+    publisher.name.1.2.3" -> { name = "publisher.name"; value = "1.2.3" }
+    ```
+  */
+  splitExt =
+    name:
+    let
+      parts = lib.splitString "." name;
+      names = lib.take 2 parts;
+      fullName = lib.concatStringsSep "." names;
+      versions = lib.drop 2 parts;
+      fullVersion = lib.concatStringsSep "." versions;
+    in
+    lib.nameValuePair (lib.toLower fullName) fullVersion;
+
+  /**
+    Indicate if the given version selector matches the vscode version.
+
+    - `^M.m.p` means `>=M.m.p`
+    - `~M.m.p` is undocumented. We assume it means `>=M.m.p && <M.(m+1).0`
+
+    DOCS https://code.visualstudio.com/api/working-with-extensions/publishing-extension#visual-studio-code-compatibility
+
+    # Type
+
+    ```
+    matchesVscodeVersion :: string -> boolean
+    ```
+
+    # Example
+
+    In the following examples, the contextual `vscodeVersion` is `"1.50.3"`:
+
+    ```nix
+    matchesVscodeVersion ">=1.50.1" -> true
+    matchesVscodeVersion ">1.50.1" -> true
+    matchesVscodeVersion "^1.50.1" -> true
+    matchesVscodeVersion "1.50.1" -> false
+    matchesVscodeVersion "~1.50.1" -> true
+
+    matchesVscodeVersion ">=1.49.1" -> true
+    mmatchesVscodeVersion ">1.49.1" -> true
+    matchesVscodeVersion "^1.49.1" -> true
+    matchesVscodeVersion "1.49.1" -> false
+    matchesVscodeVersion "~1.49.1" -> false
+
+    matchesVscodeVersion ">=0.99.1" -> true
+    matchesVscodeVersion "^0.99.1" -> false
+    matchesVscodeVersion "0.99.1" -> false
+    matchesVscodeVersion "~0.99.1" -> false
+    ```
+  */
+  matchesVscodeVersion =
```

Comment:
could you add a test for this function? just like this:
https://github.com/nix-community/nix4vscode/blob/4ff1688bee19d3443e1139436b0a015a94599862/nix/version.test.nix

Reply:
Tests added.

Reply:
Here is the source of the tests: https://github.com/microsoft/vscode/blob/747b90cfcd486731a1295563423b1ee66dd9b613/src/vs/platform/extensions/test/common/extensionValidator.test.ts

Reply:
Thanks! Indeed there were many quirks.

Included and all fixed.